### ISSUE TITLE
[Ambient] The pod can not call the K8S service.

### ISSUE THREAD

**Issue by laiminhtrung1997** [istio/istio#52216]
### Is this the right place to submit this?

- [X] This is not a security vulnerability or a crashing bug
- [X] This is not a question about how to use Istio

### Bug Description

Some pods can not call the K8S service.
```
root@tools:/# curl -Iv http://vault:8200
*   Trying 10.100.97.189:8200...
* Connected to vault (10.100.97.189) port 8200 (#0)
> HEAD / HTTP/1.1
> Host: vault:8200
> User-Agent: curl/7.88.1
> Accept: */*
> 
* Recv failure: Connection reset by peer
* Closing connection 0
curl: (56) Recv failure: Connection reset by peer
```

```
root@tools:/# curl -Iv http://kubernetes.default
*   Trying 10.100.0.1:80...
* Connected to kubernetes.default (10.100.0.1) port 80 (#0)
> HEAD / HTTP/1.1
> Host: kubernetes.default
> User-Agent: curl/7.88.1
> Accept: */*
> 
* Recv failure: Connection reset by peer
* Closing connection 0
curl: (56) Recv failure: Connection reset by peer
```

The ztunnel pod log
```
2024-07-22T07:42:09.312508Z	warn	access	connection failed	src.addr=10.100.4.18:32818 dst.addr=10.100.97.189:8200 direction="outbound" error="no healthy upstream: 10.100.97.189:8200"
2024-07-22T07:42:15.857611Z	warn	access	connection failed	src.addr=10.100.4.18:52988 dst.addr=10.100.97.189:8200 direction="outbound" error="no healthy upstream: 10.100.97.189:8200"
2024-07-22T07:45:32.764994Z	warn	access	connection failed	src.addr=10.100.4.18:34354 dst.addr=10.100.0.1:80 direction="outbound" error="no healthy upstream: 10.100.0.1:80"
2024-07-22T07:45:38.147440Z	warn	access	connection failed	src.addr=10.100.4.18:34942 dst.addr=10.100.0.1:80 direction="outbound" error="no healthy upstream: 10.100.0.1:80"
2024-07-22T07:47:36.247699Z	warn	access	connection failed	src.addr=10.100.4.18:42316 dst.addr=10.100.0.1:80 direction="outbound" error="no healthy upstream: 10.100.0.1:80"
2024-07-22T07:47:39.282199Z	warn	access	connection failed	src.addr=10.100.4.18:53900 dst.addr=10.100.97.189:8200 direction="outbound" error="no healthy upstream: 10.100.97.189:8200"
```

And the pod vault can not call kubernetes service too.
```
2024-07-22T07:48:43.484520Z	error	access	connection complete	src.addr=10.100.4.17:47246 src.workload="vault-0" src.namespace="service" dst.addr=54.239.96.26:443 direction="outbound" bytes_sent=2347 bytes_recv=6700 duration="28ms" error="io error: Connection reset by peer (os error 104)"
2024-07-22T07:48:43.594118Z	error	access	connection complete	src.addr=10.100.4.17:50964 src.workload="vault-0" src.namespace="service" dst.addr=10.113.35.124:443 dst.service="kubernetes.default.svc.cluster.local" dst.workload="kubernetes" dst.namespace="default" direction="outbound" bytes_sent=2294 bytes_recv=3928 duration="63ms" error="io error: Broken pipe (os error 32)"
```

### Version

```Text
$ istioctl version -n kube-system -i kube-system
client version: 1.22.3
control plane version: 1.22.3
data plane version: 1.22.3 (4 proxies)
```

**Comment by howardjohn** [comment]
As discussed in slack this is likely a logging issue fixed in https://github.com/istio/ztunnel/pull/1219

**Comment by laiminhtrung1997** [comment]
> As discussed in slack this is likely a logging issue fixed in https://github.com/istio/ztunnel/pull/1219

So I continue waiting for the release 1.22 :( Hope that it'll release soon.

**Comment by howardjohn** [comment]
@laiminhtrung1997 on slack I sent an image with the bug fix if you want to try it out.

**Comment by laiminhtrung1997** [comment]
@howardjohn The image works very well. I am going to continue to test the ambient. Thanks.

**Comment by laiminhtrung1997** [comment]
The log from the pod called vault service
```
Error checking seal status: Get "http://vault:8200/v1/sys/seal-status": EOF
Error checking seal status: Get "http://vault:8200/v1/sys/seal-status": read tcp 10.100.198.188:39612->10.100.217.195:8200: read: connection reset by peer
```

The log from ztunnel
```
2024-07-23T05:03:55.903524Z	warn	access	connection failed	src.addr=10.100.198.188:53422 dst.addr=10.100.217.195:8200 direction="outbound" error="no healthy upstream: 10.100.217.195:8200"
```

I am still facing the issue when calling the vault service.

**Comment by sachintiptur** [comment]
We also have a similar problem with may applications in Ambient mode failing because they can't call K8s API server. 

```

2024-07-25T08:56:18.362284Z     warn    access  connection failed       src.addr=10.233.109.149:42700 dst.addr=10.233.0.1:443 direction="outbound" error="no healthy upstream: 10.233.0.1:443"
2024-07-25T08:56:18.362324Z     warn    access  connection failed       src.addr=10.233.109.149:42712 dst.addr=10.233.0.1:443 direction="outbound" error="no healthy upstream: 10.233.0.1:443"
2024-07-25T08:56:18.363914Z     warn    access  connection failed       src.addr=10.233.109.131:33370 dst.addr=10.233.0.1:443 direction="outbound" error="no healthy upstream: 10.233.0.1:443"
2024-07-25T08:56:18.373788Z     warn    access  connection failed       src.addr=10.233.109.136:60662 dst.addr=10.233.0.1:443 direction="outbound" error="no healthy upstream: 10.233.0.1:443"
```

While debugging we noticed some ssl error for grafana related pods where ssl errors were seen, once disabling ssl for those pods, it worked.(We did this by anyways Istio does mTLS for all traffic in the mesh). But for rest of the services in mesh we could not fix. All have the same signature of errors like unable to make call to k8s api-server. For ex:, calico-controller is in crashloopbackoff once ambient mode is enabled, similarly few other pods too.

**Comment by sachintiptur** [comment]
I hope [this](https://github.com/istio/istio/pull/52191) would solve this problem

**Comment by howardjohn** [comment]
@sachintiptur what version? that issue should be resolved

**Comment by howardjohn** [comment]
fix is in 1.22.3

**Comment by laiminhtrung1997** [comment]
@howardjohn 
How about the issue that I mentioned on Slack confirmed by you? Is there a PR for it?

**Comment by sachintiptur** [comment]
> fix is in 1.22.3

Tested with this version, but still the same error.
For testing, we are enabling ambient mesh only for one service(kube-prometheus-stack). These errors are seen for some of the pods in this service.

**Comment by howardjohn** [comment]
@sachintiptur can you upload an `istioctl bug-report`? feel free to DM it on slack
